include/zephyr/devicetree.h

```diff
@@ -4557,10 +4557,6 @@
 #define DT_NODE_HAS_STATUS_INTERNAL(node_id, status) \
 	IS_ENABLED(DT_CAT3(node_id, _STATUS_, status))
 
-/** @brief Helper macro to OR multiple has property checks in a loop macro */
-#define DT_INST_NODE_HAS_PROP_AND_OR(inst, prop) \
```

Comment:
This is a public macro. We can deprecate it, but shouldn't just remove it. 

If you go with deprecation, please update the release notes as well.

Reply:
Is it considered a public macro if it's undocumented? Note the `@cond INTERNAL_HIDDEN` further up.
If anyone actually uses it, then I'd rather rewrite it than remove it, but I think it's just a leftover.

Reply:
Ah right. False alarm, sorry.

Reply:
And looking at the definition, it is very much just a helper macro